design-proposals/storage-class-migration/storage-class-migration.md

```diff
@@ -0,0 +1,276 @@
+If the user specifies the storage migration classes, then the destination PVCs will be created before updating the VMI status.
+
+Afterwards, the controller will create a VirtualMachineInstanceMigration to trigger the VM live migration.
+
+The controller if there are some migrated volume in the VMI status, it will replace the destination volumes by creating the target virt-launcher pod and virt-launcher will select the source volume to be migrated.
+
+Once the migration has succeeded, the controller will update firstly the migrated volume under the VMI specification. This update will consequentally trigger the update of the volume in the VM.
+
+DataVolumes are taken into account in the current proposal. DataVolumes are used to import the image during VM provisioning, but they may essentially be thought of as PVCs for the rest of the VM life cycle. According to this proposal, KubeVirt will replace the source DataVolumes with the destination PVC when a user wishes to migrate data to a PVC.
+
+A schematic representation of the migration flow is shown below.
+![](storage-class-mig.png)
+
+### API examples
+
+Example of StorageMigration
+
+Create a VM with source storage and the destination PVC:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: dest-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  annotations:
+    kubevirt.io/libvirt-log-filters: "1:qemu.qemu_monitor 1:*"
+  labels:
+    kubevirt.io/vm: vm-alpine-datavolume
+  name: vm-alpine-datavolume
+spec:
+  dataVolumeTemplates:
+  - metadata:
+      creationTimestamp: null
+      name: source-pvc
+    spec:
+      pvc:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: local
+      source:
+        registry:
+          url: docker://registry:5000/kubevirt/alpine-container-disk-demo:devel
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-alpine-datavolume
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      networks:
+      - name: default
+        pod: {}
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - dataVolume:
+          name: source-pvc
+        name: datavolumedisk1
+
+```
+
+Trigger the migration with:
+```yaml
+apiVersion: storage.kubevirt.io/v1alpha1
+kind: StorageMigration
+metadata:
+  name: sc-mig
+spec:
+  vmiName: vmi
+  migratedVolume:
+  - sourcePvc: source-pvc
+    destinationPvc: dest-pvc
```

Comment:
Given the question above about making the `MigrationVolume` a triple this should get updated.

Reply:
I'll update based on the discussion above, please check my comment